api/v1alpha1/freight_types.go

```diff
@@ -172,10 +174,24 @@ type FreightStatus struct {
 	ApprovedFor map[string]ApprovedStage `json:"approvedFor,omitempty" protobuf:"bytes,2,rep,name=approvedFor" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
 }
 
+// CurrentStage reflects a Stage's current use of Freight.
+type CurrentStage struct {
+	// Since is the time at which the Stage most recently started using the
+	// Freight. This can be used to calculate how long the Freight has been in use
+	// by the Stage.
+	Since *metav1.Time `json:"since,omitempty" protobuf:"bytes,1,opt,name=since"`
+}
+
 // VerifiedStage describes a Stage in which Freight has been verified.
 type VerifiedStage struct {
 	// VerifiedAt is the time at which the Freight was verified in the Stage.
 	VerifiedAt *metav1.Time `json:"verifiedAt,omitempty" protobuf:"bytes,1,opt,name=verifiedAt"`
+	// LongestCompletedSoak represents the longest definite time interval wherein
+	// the Freight was in CONTINUOUS use by the Stage. This value is updated as
+	// Freight EXITS the Stage. If the Freight is currently in use by the Stage,
+	// the time elapsed since the Freight ENTERED the Stage is its current soak
+	// time, which may exceed the value of this field.
+	LongestCompletedSoak *metav1.Duration `json:"longestSoak,omitempty" protobuf:"bytes,2,opt,name=longestSoak"`
```

Comment:
I can not help but feel the placement of this is odd, and if we would be better off keeping track of "entering" / "leaving" history to compute this value.

Reply:
I had this exact thought, but had trouble justify all the additional bookkeeping because we really only care about this info for Stages in which the Freight has been verified.

Were we to track entering/leaving separately, it'd also be one more thing to clean up during finalization of a deleted Freight -- which isn't a _huge_ deal.

Consider me completely amenable to doing this differently if you can suggest a way that isn't too onerous and doesn't create an accumulation of unimportant data.

Reply:
We took this offline and struggled to come up with anything that was obviously better. The one potential thing we came up with wasn't better enough to justify the migration logic and deprecated fields that would have to stick around for many released to come. We agreed to come back to this someday if something worth paying that price for presents itself.